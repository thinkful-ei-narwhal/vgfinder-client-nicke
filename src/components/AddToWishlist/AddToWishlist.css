.add-button {
  margin-top: 15px;
  margin-bottom: 15px;
  color: black;
  background-color: lightgray;
}

.wishlisted-color {
  background-color: #5dc21e;
  border: none;
  color: white;
}
