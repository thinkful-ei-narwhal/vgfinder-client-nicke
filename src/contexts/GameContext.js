import React, { Component } from "react";

export const nullGame = {
  id: "",
  title: "",
  description: "",
  genre: "",
  rating: "",
  release_date: "",
  developer: "",
  trailer_url: "",
  image_url_box_art: "",
  image_url_two: "",
  image_url_three: "",
  image_url_four: "",
  image_url_five: "",
};

const GameContext = React.createContext({
  game: nullGame,
  setGame: () => {},
  clearGame: () => {},
});

export default GameContext;

export class GameProvider extends Component {
  state = {
    game: nullGame,
  };

  setGame = (game) => {
    this.setState({ game });
  };

  clearGame = () => {
    this.setGame(nullGame);
  };

  render() {
    const value = {
      game: this.state.game,
      setGame: this.setGame,
      clearGame: this.clearGame,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}
