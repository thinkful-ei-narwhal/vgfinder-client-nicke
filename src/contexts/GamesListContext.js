import React, { Component } from 'react';

const GamesListContext = React.createContext({
  gamesList: [],
  error: null,
  setError: () => { },
  clearError: () => { },
  setGamesList: () => { },
})
export default GamesListContext

export class GamesListProvider extends Component {
  state = {
    gamesList: [],
    error: null,
  };

  setGamesList = gamesList => {
    this.setState({ gamesList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      gamesList: this.state.gamesList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGamesList: this.setGamesList,
    }
    return (
      <GamesListContext.Provider value={value}>
        {this.props.children}
      </GamesListContext.Provider>
    )
  }
}
