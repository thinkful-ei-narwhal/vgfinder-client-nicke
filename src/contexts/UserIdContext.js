import React, { Component } from 'react'

const UserIdContext = React.createContext({
  userId: undefined,
  wishlist: [],
  setUserId: () => { },
  setUserWishlist: () => { },
})

export default UserIdContext

export class UserIdContextProvider extends Component {
  state = {
    userId: undefined,
    wishlist: []
  };

  setUserId = (userId) => {
    this.setState({ userId })
  }

  setUserWishlist = (wishlist) => {
    this.setState({ wishlist })
  }


  render() {
    const value = {
      userId: this.state.userId,
      wishlist: this.state.wishlist,
      setUserId: this.setUserId,
      setUserWishlist: this.setUserWishlist
    }

    return (
      <UserIdContext.Provider value={value}>
        {this.props.children}
      </UserIdContext.Provider>
    )
  }
}
