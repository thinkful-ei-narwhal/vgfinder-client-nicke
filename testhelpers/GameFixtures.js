const GameFixtures = {
  stubGameData: {
    id: "1",
    title: "Hollow Knight",
    description:
      "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
    genre: "Adventure",
    rating: "4",
    developer: "Team Cherry",
    trailer_url:
      "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
    image_url_box_art:
      "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
    image_url_two:
      "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
    image_url_three:
      "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
  },
  stubGameListData: [
    {
      id: "1",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
    {
      id: "2",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
  ],
};

export default GameFixtures;
