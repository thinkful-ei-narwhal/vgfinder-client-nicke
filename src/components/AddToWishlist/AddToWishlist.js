
import './AddToWishlist.css'
import React, { Component } from 'react'
import UserIdContext from '../../contexts/UserIdContext';
import WishlistService from '../../services/wishlist-api-service'

export default class AddToWishlist extends Component {
  static contextType = UserIdContext;
  state = {
    loaded: false,
  }

  handleAddToWishlist = () => {
    const userId = this.context.userId
    const game = this.props.game

    WishlistService.addToWishlist(userId, game.id)
      .then(res => {
        const wishlist = this.context.wishlist;
        wishlist.push(res);

        const userGames = this.context.userGames;
        userGames.push(game);

        this.context.setUserIdWishlistAndGames(userId, wishlist, userGames);
      });
  }

  componentDidMount() {
    this.setState({ loaded: true });
  }

  renderButton() {
    const isInWishlist = !!this.context.wishlist.find(wishlistItem => wishlistItem.game_id === this.props.game.id);
    return (
      <button onClick={this.handleAddToWishlist} disabled={isInWishlist}>
        {isInWishlist ? "Wishlisted" : "Add to Wishlist"}
      </button>
    )
  }

  render() {
    return (
      this.state.loaded ? this.renderButton() : null
    )
  }
}
