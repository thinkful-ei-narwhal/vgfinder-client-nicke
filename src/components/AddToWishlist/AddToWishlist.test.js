import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddToWishlist from './AddToWishlist'
import { UserIdContextProvider } from './../../contexts/UserIdContext'
import GameFixtures from './../../../testhelpers/GameFixtures'


describe(`Add to wishlist button component`, () => {
  it('renders the Add to wishlist button if you supply a game', () => {
    const wrapper = shallow(
      <UserIdContextProvider >
        <AddToWishlist game={GameFixtures.stubGameData} />
      </UserIdContextProvider >
    )
      .find('#note-folder-select')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
