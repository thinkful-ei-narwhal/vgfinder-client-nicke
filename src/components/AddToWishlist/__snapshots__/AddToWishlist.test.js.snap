// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add to wishlist button component renders the Add to wishlist button if you supply a game 1`] = `null`;
